imu_sensor_controller:
  type: imu_sensor_controller/ImuSensorController
  publish_rate: 1000


# Publish all joint states -----------------------------------
# 100Hz is a compromise between fast reaction and CPU load
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 1000

DynamixelController:
  type: bitbots_ros_control/DynamixelController
  joints:
      - RShoulderPitch
      - LShoulderRoll
      - RShoulderRoll


TestController:
  type: position_controllers/JointPositionController
  joint: No4
  pid: {p: 50.0, i: 0.00, d: 0.4}

TestController2:
  type: effort_controllers/JointPositionController
  joint: HeadPan  
  pid: {p: 50.0, i: 0.00, d: 0.4}

JointGroupController:
  type: position_controllers/JointGroupPositionController
  joints:
      - HeadPan
      - HeadTilt
      - LShoulderPitch
      - LShoulderRoll
      - LElbow
      - RShoulderPitch
      - RShoulderRoll
      - RElbow
      - LHipYaw
      - LHipRoll
      - LHipPitch
      - LKnee
      - LAnklePitch
      - LAnkleRoll
      - RHipYaw
      - RHipRoll
      - RHipPitch
      - RKnee
      - RAnklePitch
      - RAnkleRoll

JointGroupControllerEffort:
  type: effort_controllers/JointGroupPositionController
  joints: 
    - HeadPan
    - HeadTilt
    - LShoulderPitch
    - LShoulderRoll
    - LElbow
    - RShoulderPitch
    - RShoulderRoll
    - RElbow
    - LHipYaw
    - LHipRoll
    - LHipPitch
    - LKnee
    - LAnklePitch
    - LAnkleRoll
    - RHipYaw
    - RHipRoll
    - RHipPitch
    - RKnee
    - RAnklePitch
    - RAnkleRoll
  HeadPan:        { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  HeadTilt:       { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LShoulderPitch: { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LShoulderRoll:  { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LElbow:         { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RShoulderPitch: { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RShoulderRoll:  { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RElbow:         { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LHipYaw:        { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LHipRoll:       { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LHipPitch:      { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LKnee:          { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  LAnklePitch:    { pid: { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 } }
  LAnkleRoll:     { pid: { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 } }
  RHipYaw:        { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RHipRoll:       { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RHipPitch:      { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RKnee:          { pid: { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 } }
  RAnklePitch:    { pid: { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 } }
  RAnkleRoll:     { pid: { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 } }


TrajectoryController:
  type: position_controllers/JointTrajectoryController
  constraints/goal_time: 1.0
  allow_partial_joints_goal: True
  joints:
      - HeadPan
      - HeadTilt
      - LShoulderPitch
      - LShoulderRoll
      - LElbow
      - RShoulderPitch
      - RShoulderRoll
      - RElbow
      - LHipYaw
      - LHipRoll
      - LHipPitch
      - LKnee
      - LAnklePitch
      - LAnkleRoll
      - RHipYaw
      - RHipRoll
      - RHipPitch
      - RKnee
      - RAnklePitch
      - RAnkleRoll
  gains:
    No1:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No2:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No3:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No4:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No5:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No6:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No7:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No8:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No9:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No10:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No11:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No12:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No13:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No14:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No15:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No16:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No17:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No18:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No19:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }
    No20:        { p: 50.0, i: 0.00, d: 0.4, i_clamp: 4.0 }


TrajectoryControllerEffort:
  type: effort_controllers/JointTrajectoryController
  constraints/goal_time: 1.0
  allow_partial_joints_goal: True
  joints: 
    - HeadPan
    - HeadTilt
    - LShoulderPitch
    - LShoulderRoll
    - LElbow
    - RShoulderPitch
    - RShoulderRoll
    - RElbow
    - LHipYaw
    - LHipRoll
    - LHipPitch
    - LKnee
    - LAnklePitch
    - LAnkleRoll
    - RHipYaw
    - RHipRoll
    - RHipPitch
    - RKnee
    - RAnklePitch
    - RAnkleRoll
  gains:
    HeadPan:         { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    HeadTilt:        { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LShoulderPitch:  { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LShoulderRoll:   { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LElbow:          { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RShoulderPitch:  { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RShoulderRoll:   { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RElbow:          { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LHipYaw:         { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LHipRoll:        { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LHipPitch:       { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LKnee:           { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    LAnklePitch:     { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 }
    LAnkleRoll:      { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 }
    RHipYaw:         { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RHipRoll:        { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RHipPitch:       { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RKnee:           { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
    RAnklePitch:     { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 }
    RAnkleRoll:      { p: 5.0, i: 0.00, d: 0.2, i_clamp: 0.5 } 



HeadTiltTrajectoryControllerEffort:
  type: effort_controllers/JointTrajectoryController
  constraints/goal_time: 1.0
  allow_partial_joints_goal: True
  joints: 
    - HeadTilt
  gains:
    HeadTilt:        { p: 50.0, i: 0.00, d: 1.0, i_clamp: 4.0 }